_local.tf

```diff
@@ -6,22 +6,21 @@ resource "random_string" "s3_viewer" {
 
 locals {
   registry_url      = var.disable_ecr == false ? "${data.aws_caller_identity.current.account_id}.dkr.ecr.${var.region}.amazonaws.com" : "ghcr.io/cdcgov/phdi"
-  registry_auth     = data.aws_ecr_authorization_token.this.proxy_endpoint
   registry_username = data.aws_ecr_authorization_token.this.user_name
   registry_password = data.aws_ecr_authorization_token.this.password
-  service_data = {
+  service_data = var.service_data == {} ? {
```

Comment:
Small item, non-blocking: would it be more legible for us to use the `coalesce` function here, so the last part of the ternary isn't dangling off the end of the block a few dozen lines down?

More info here: https://developer.hashicorp.com/terraform/language/functions/coalesce

Reply:
Hey! I looked at `coalesce,` which seems like a good win for readability. 🔺 

I also took a peak at `merge`, I could see using it to allow for more refined control of the service_data object without needing to pass in the entire blob. ( like maybe I just want the orchestration service to have more memory) That said, it will not work for today, so I'll file that away in the maybe someday column.

Reply:
After testing coalesce, I got some very weird behavior. The most consistent behavior came from checking the `service_data` object with this: 
```
length(var.service_data) > 0 ? var.service_data : ...
```
I'd like to use this version; it's consistent and solves the issue of `var.service_data` getting lost at the end.